projects/extension/sql/idempotent/900-semantic-catalog-init.sql

```diff
@@ -0,0 +1,69 @@
+--FEATURE-FLAG: text_to_sql
+
+-------------------------------------------------------------------------------
+-- initialize_semantic_catalog
+create or replace function ai.initialize_semantic_catalog
+( "name" pg_catalog.name default 'default'
+, embedding pg_catalog.jsonb default null
+, indexing pg_catalog.jsonb default ai.indexing_default()
+, scheduling pg_catalog.jsonb default ai.scheduling_default()
+, processing pg_catalog.jsonb default ai.processing_default()
+, grant_to pg_catalog.name[] default ai.grant_to()
+) returns pg_catalog.int4
+as $func$
+declare
+    _catalog_id pg_catalog.int4;
+    _obj_vec_id pg_catalog.int4;
+    _sql_vec_id pg_catalog.int4;
+begin
+    grant_to = pg_catalog.array_cat
+    ( grant_to
+    , array
+      [ pg_catalog."session_user"()
+      , 'pg_database_owner'::name
+      ]
+    );
+
+    insert into ai.semantic_catalog("name")
+    values (initialize_semantic_catalog."name")
+    returning id
+    into strict _catalog_id
+    ;
+
+    select ai.create_vectorizer
+    ( 'ai.semantic_catalog_obj'::pg_catalog.regclass
+    , destination=>pg_catalog.format('semantic_catalog_obj_%s', _catalog_id)
+    , embedding=>embedding
+    , indexing=>indexing
+    , scheduling=>scheduling
+    , processing=>processing
+    , grant_to=>grant_to
+    , formatting=>ai.formatting_python_template() -- TODO: this ain't gonna work
+    , chunking=>ai.chunking_recursive_character_text_splitter('description') -- TODO
```

Comment:
What's still required here?

Reply:
If I implement custom python formatting, then it can stay as is. If I have to create another column and use a trigger to do my formatting, then the chunking will need to happen on that column.